packages/react-kit/src/components/Selectbox/Selectbox.tsx

```diff
@@ -106,7 +107,24 @@ class RawSelectbox extends React.Component<TFullSelectboxProps, TSelectboxState>
 		}
 	}
 
+	closeOnScroll = (event: UIEvent) => {
+		if (this.anchor) {
+			const element = findDOMNode(this.anchor);
+			if (!element || element instanceof Text) {
+				return;
+			}
+
+			if (event.target && !element.contains(event.target as Node)) {
+				this.setState({ isOpened: false });
+			}
+		}
+	};
+
 	componentDidMount() {
```

Comment:
What about `componentWillReceiveProps`? You can use [EventListener](https://github.com/devex-web-frontend/dx-platform/blob/master/packages/react-kit/src/components/EventListener/EventListener.tsx) here.